VSConfigFinder/FileSystem.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace VSConfigFinder
+{
+    using System;
+    using System.ComponentModel.DataAnnotations;
+    using System.Diagnostics.CodeAnalysis;
+    using System.IO;
+    using System.Text.Json;
+
+    public class FileSystem : IFileSystem
+    {
+        /// <inheritdoc/>
+        public string[] GetFileSystemEntries(string path, string pattern, bool recursive = false)
+        {
+            Utilities.ValidateIsNotNullOrEmpty(path, nameof(path));
+            Utilities.ValidateIsNotNullOrEmpty(pattern, nameof(pattern));
+
+            return Directory.GetFileSystemEntries(path, pattern, recursive ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly);
+        }
+
+        /// <inheritdoc/>
+        public Stream OpenFile(string path)
+        {
+            Utilities.ValidateIsNotNullOrEmpty(path, nameof(Path));
+
+            return File.OpenRead(path);
```

Comment:
It'd be great to merge `CreateOutput` utility method where it calls `File.WriteAllText` into this new class. What do you think? (Even if it's just a one-liner plus some validation)

Reply:
Like in FileSystem.cs, create method that does the if block logic in CreateOutput? If so, I think it makes sense since that block does File stuff.

Reply:
Yep, if you look at our Setup code we wrap all the `File.xx` calls with the FileSystem calls so that it can be testable with mocks. If you could do that while you're on it, it'd be great!